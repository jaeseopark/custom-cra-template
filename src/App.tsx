import { useEffect, useState } from "react";
import { useDispatch } from "react-redux";
import styled from "styled-components";

import { initializeClient } from "redux/mdlwr";
import ChatContainer from "component/chat/ChatContainer";
import ConnectivityProvider from "component/ConnectivityProvider";
import NotificationProvider from "component/NotificationProvider";
import PreviewCardContainer from "component/preview/PreviewCardContainer";
import SearchBar from "component/SearchBar";
import { APPLE_BIGSUR_GRAY_BACKGROUND, APPLE_BIGSUR_GRAY_OUTLINE, BOX_SHADOW_COLOR } from "style/const";
import Settings from "component/Settings";

const APP_MARGIN = 40; // pixels

const StyledApp = styled.div`
    display: flex;
    flex-flow: row nowrap;
    height: calc(100vh - ${APP_MARGIN}px);
    width: calc(100vw - ${APP_MARGIN}px);
    box-shadow: 0px 10px 35px ${BOX_SHADOW_COLOR};
    border-radius: 13px;
    overflow: hidden;
    margin: ${APP_MARGIN / 2}px;
    background-color: ${APPLE_BIGSUR_GRAY_BACKGROUND};
`;

const Sidebar = styled.div`
    display: flex;
    flex-flow: column nowrap;
    width: 350px;
    padding: 0 10px 0 10px;
    border-right: 1px solid ${APPLE_BIGSUR_GRAY_OUTLINE};
    /* resize: horizontal; */
`;

const App = () => {
    const dispatch = useDispatch();
    const [selectedAlias, selectAlias] = useState<string>();

    useEffect(() => {
        dispatch(initializeClient());
    }, [dispatch]);

    return (
        <StyledApp>
            <Sidebar>
                <Settings />
                <SearchBar />
                <PreviewCardContainer onClickAlias={selectAlias} selectedAlias={selectedAlias} />
            </Sidebar>
            <ChatContainer alias={selectedAlias} />
        </StyledApp>
    );
};

export default App;
