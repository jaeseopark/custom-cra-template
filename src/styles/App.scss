.App {
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
}

.chatContainer {
	display: flex;
	flex-direction: column;
	justify-content: center;
	height: fit-content;
	width: 50%;
	margin: 3rem;
	border: 0.1rem solid black;
}

.contactContainer {
	min-height: 10rem;
	height: fit-content;

	.chatInput {
		width: 85%;
	}

	.textInput {
		width: 100%;
	}

	.timeStamp {
		color: gray;
		margin-right: 1rem;
		white-space: nowrap;
	}

	.textContainer {
		display: flex;
		justify-content: space-between;
		padding: 0.2rem 0.7rem;
		width: 85%;

		.textInput {
			width: 100%;
		}

		.timeStamp {
			align-self: center;
			color: gray;
			margin-right: 1rem;
			white-space: nowrap;
		}
	}
}
