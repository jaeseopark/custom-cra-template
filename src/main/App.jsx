import React from "react";

import ChatContainer from "src/components/ChatContainer";

import "src/styles/App.scss";

const App = () => {
	return (
		<div className="App">
			<ChatContainer />
		</div>
	);
};

export default App;
