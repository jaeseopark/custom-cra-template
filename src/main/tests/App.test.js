import React from "react";
import { render } from "@testing-library/react";
import App from "../App";

test("it renders the app", () => {
  const { container, getByText } = render(<App />);
  expect(getByText("Edit")).toBeTruthy();
  expect(container).toMatchSnapshot();
});
