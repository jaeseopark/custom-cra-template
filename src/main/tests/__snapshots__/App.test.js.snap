// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the app 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <img
        alt="logo"
        class="App-logo"
        src="test-file-stub"
      />
      <p>
        Edit
      </p>
      <a
        class="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
  </div>
</div>
`;
